metals/src/main/scala/scala/meta/internal/metals/ServerLivenessMonitor.scala

```diff
@@ -61,6 +62,7 @@ class ServerLivenessMonitor(
     serverName: String,
     metalsIdleInterval: Duration,
     pingInterval: Duration,
+    restartBsp: () => Future[Boolean],
```

Comment:
Actually, if we are not connected to build server this will not work, my bad for suggesting it. We should show `connect to build server instead` `reestablishConnection` in BuildServerConnection.

There also seems to be an issue where we don't shutdown the monitor for older connection when restarting, so this would cause this issue to be bigger.

Reply:
I added `disconnect` before shutting down bloop and it seems to help.
Plus I changed `restartBsp` to `reestablishConnection` but now I that I'm thinking, I'm not sure why. If the sever is not responsive shouldn't we try to kill it?